src/main/java/Wolox/training/controllers/BookController.java

```diff
@@ -45,6 +58,52 @@ public Book findById(@PathVariable int id) {
         return bookRepository.findById(id).orElseThrow(() -> new RuntimeException("The book does not exist"));
     }
 
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public Book notFound() {
+        return null;
+    }
+
+    @ResponseStatus(HttpStatus.OK)
+    @GetMapping
+    public Book findLocalByIsbn(String isbn) {
+        return bookRepository.findByIsbn(isbn).orElseThrow(() -> new NoSuchElementException());
+    }
+
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping
+    public Book findInOnlineLibrary(String isbn) {
+        try {
+            BookDAO bookDAO = onlineLibrary.bookInfo(isbn);
+            Book book = new Book();
```

Comment:
you can add a constructor in the book method to take a bookDAO and return a created book.
Better add that code to the book model and not set all the properties in the controller method